server/build/babel/preset.js

```diff
@@ -30,14 +40,14 @@ module.exports = {
       require.resolve('babel-plugin-module-resolver'),
       {
         alias: {
-          'babel-runtime': babelRuntimePath,
-          'next/link': require.resolve('../../../lib/link'),
-          'next/prefetch': require.resolve('../../../lib/prefetch'),
-          'next/css': require.resolve('../../../lib/css'),
-          'next/head': require.resolve('../../../lib/head'),
-          'next/document': require.resolve('../../../server/document'),
-          'next/router': require.resolve('../../../lib/router'),
-          'next/error': require.resolve('../../../lib/error')
+          'babel-runtime': packageRelativeResolve('babel-runtime/package'),
+          'next/link': packageRelativeResolve('../../../lib/link'),
+          'next/prefetch': packageRelativeResolve('../../../lib/prefetch'),
+          'next/css': packageRelativeResolve('../../../lib/css'),
+          'next/head': packageRelativeResolve('../../../lib/head'),
+          'next/document': packageRelativeResolve('../../../server/document'),
+          'next/router': packageRelativeResolve('../../../lib/router'),
+          'next/error': packageRelativeResolve('../../../lib/error')
```

Comment:
By stripping everything up to (and including) the first `node_modules/` they end up being relative to the top-level NPM root.

That being said, a better solution may be to enable the old method only for local tests and peerDependency in the other things (based on the discussion in #1160)